lib/spack/spack/environment/environment.py

```diff
@@ -876,6 +912,8 @@ def clear(self, re_read=False):
         self.concretized_user_specs = []  # user specs from last concretize
         self.concretized_order = []  # roots of last concretize, in order
         self.specs_by_hash = {}  # concretized specs by hash
+        self.included_concretized_user_specs = []  # user specs from last concretize's included env
+        self.included_concretized_order = []  # roots of the included envs, in order
```

Comment:
I think this makes the `Environment` class even less coherent than it is now, in the sense that it will have more attributes and will use just a few of them in most methods. This is usually an indicator that the class has too many responsibilities.

Also, these attributes are set outside of `__init__`. To be fair, they're not the first ones like that for `Environment`, but in #36927 I made sure to declare every attribute of an environment in the constructor.

Reply:
I agree that there are too many properties/responsibilities but included equivalents were added to the constructor at some point since your comment.